src/odin/adapters/parameter_tree.py

```diff
@@ -321,23 +313,41 @@ def set(self, path, data):
         else:
             merge_parent[int(levels[-1])] = merged
 
-    def add_callback(self, path, callback):
-        """Add a callback to a given path in the tree - DEPRECATED.
+    def delete(self, path=''):
+        """
+        Remove Parameters from a Mutable Tree.
 
-        This now deprecated method adds a callback to the specified path in the
-        tree. Originally intended to allow set() calls to update values in the
-        underlying object or device represented by the tree, this has been
-        replaced by the symmetric read/write ParameterAccessor mechanism. Its
-        remaining function could be to allow side-effects during set() calls.
+        This method deletes selected parameters from a tree, if that tree has been flagged as
+        Mutable. Deletion of Branch Nodes means all child nodes of that Branch Node are also deleted
 
-        :param path: path to add callback for
-        :param callback: method to be called when the appropriate set() call is made
+        :param path: Path to selected Parameter Node in the tree
         """
-        warnings.warn(
-            "Callbacks in parameter trees are deprecated, use parameter accessors instead",
-            DeprecationWarning
-        )
-        self._callbacks.append([path, callback])
+        if not self.mutable:
+            raise ParameterTreeError("Invalid Delete Attempt: Tree Not Mutable")
+        
+        # Split the path by levels, truncating the last level if path ends in trailing slash
+        levels = path.split('/')
+        if levels[-1] == '':
+            del levels[-1]
+
+        subtree = self._tree
+
+        if not levels:
+            for key in subtree.copy():  # copy to avoid runtime error
+                subtree.pop(key)
+            return
+        try:
+            for level in levels[:-1]:
+                if isinstance(subtree, dict):
+                    subtree = subtree[level]
+                else:
+                    subtree = subtree[int(level)]
```

Comment:
Could you just explain this part a little to me?  You traverse down the tree and assume if the subtree isn't a dict then the next path level is always going to be an integer, is my understanding right?

Reply:
Pretty much, yeah. This was based off the traversal stuff in the set method, which does the same thing. The idea being that, if you're still traversing the path you set, you're either:
at a branch node of the tree, thus the next part should be another dict
at a node that contains a list, thus the next part of the path should be the list index
Then, obviously, if its neither of these things, you're path is invalid and it raises the ParameterTreeError

Reply:
I think some comments dotted throughout the code explaining these details at each step would be useful